### ISSUE TITLE
How to use Cotp in TCP Socket

### ISSUE THREAD

**Issue by mahsamohammadi17** [mz-automation/libiec61850#240]
I want to write a program that receives data from the simulator of a IEC 61850.
I know that after connecting the socket, a COTP must be established. But I don't know how to define COTP connection in my socket. Can anyone help me?

**Comment by mzillgith** [comment]
I don't understand your point. Why can't you use the library?

**Comment by mahsamohammadi17** [comment]
I can use the library.
I tested between the two IEDScout emulators and recorded the exchanges with Wireshark, after the TCP connection was established, the COTP connection was established.
I used the IEC61850 library's cotp.c program to do this, but I don't know how to call it, and I don't know if I chose it correctly or if I should use another library program to call COTP.
(I am writing the server side. Of course, in some of the said posts, the server port must be 102, which I cannot use because this port is reserved.)

**Comment by Masih-tech** [comment]
I can use the library.
I tested between the two IEDScout emulators and recorded the exchanges with WireShack, after the TCP connection was established, the COTP connection was established.
I used the IEC61850 library's cotp.c program to do this, but I don't know how to call it, and I don't know if I chose it correctly or if I should use another library program to call COTP.
(I am writing the server side. Of course, in some of the said posts, the server port must be 102, which I cannot use because this port is reserved.)

**Comment by Masih-tech** [comment]
I can use the library.
I tested between the two IEDScout emulators and recorded the exchanges with WireShack, after the TCP connection was established, the COTP connection was established.
I used the IEC61850 library's cotp.c program to do this, but I don't know how to call it, and I don't know if I chose it correctly or if I should use another library program to call COTP.
(I am writing the server side. Of course, in some of the said posts, the server port must be 102, which I cannot use because this port is reserved.)

**Comment by Masih-tech** [comment]
I can use the library.
I tested between the two IEDScout emulators and recorded the exchanges with WireShack, after the TCP connection was established, the COTP connection was established.
I used the IEC61850 library's cotp.c program to do this, but I don't know how to call it, and I don't know if I chose it correctly or if I should use another library program to call COTP.
(I am writing the server side. Of course, in some of the said posts, the server port must be 102, which I cannot use because this port is reserved.)

**Comment by Masih-tech** [comment]
I can use the library.
I tested between the two IEDScout emulators and recorded the exchanges with WireShack, after the TCP connection was established, the COTP connection was established.
I used the IEC61850 library's cotp.c program to do this, but I don't know how to call it, and I don't know if I chose it correctly or if I should use another library program to call COTP.
(I am writing the server side. Of course, in some of the said posts, the server port must be 102, which I cannot use because this port is reserved.)

**Comment by Masih-tech** [comment]
I can use the library.
I tested between the two IEDScout emulators and recorded the exchanges with WireShack, after the TCP connection was established, the COTP connection was established.
I used the IEC61850 library's cotp.c program to do this, but I don't know how to call it, and I don't know if I chose it correctly or if I should use another library program to call COTP.
(I am writing the server side. Of course, in some of the said posts, the server port must be 102, which I cannot use because this port is reserved.)
